orderbook/src/orderbook.rs

```diff
@@ -267,6 +212,10 @@ impl Orderbook {
         set_available_balances(orders.as_mut_slice(), &self.solvable_orders);
         Ok(orders)
     }
+
+    fn bad_token_detector(&self) -> &dyn BadTokenDetecting {
+        self.post_order_validator.bad_token_detector.as_ref()
+    }
```

Comment:
This is why `bad_token_detector` was made into a public field inside the `PostOrderValidator`. I suspect this is kinda hacky and should fall back on just providing a  bad token detector directly to the orderbook, since it is clearly needed for more than just order validation. Its an Arc anyway, so It shouldn't be so costly to just give Orderbook its own bad token detector.

Would like to hear back from reviewers if my suspicion is on point.

Reply:
I prefer it if Orderbook stores its own Arc<BadTokenDetecting> and the field is private.